ouroboros-consensus-test/test-consensus/Test/Consensus/Mempool/Fairness.hs

```diff
@@ -0,0 +1,207 @@
+{-# LANGUAGE LambdaCase         #-}
+{-# LANGUAGE NamedFieldPuns     #-}
+{-# LANGUAGE NumericUnderscores #-}
+
+-- | Tests fairness aspects of the mempool.
+--
+-- See 'testTxSizeFairness' for more details on the tests we run in this module.
+module Test.Consensus.Mempool.Fairness (
+    testTxSizeFairness
+  , tests
+  ) where
+
+import qualified Cardano.Slotting.Time as Time
+import           Control.Arrow ((***))
+import           Control.Concurrent (threadDelay)
+import qualified Control.Concurrent.Async as Async
+import           Control.Monad (forever, void)
+import qualified Control.Tracer as Tracer
+import           Data.Foldable (asum)
+import qualified Data.List as List
+import           Data.Word (Word32)
+import           Ouroboros.Consensus.Config.SecurityParam as Consensus
+import qualified Ouroboros.Consensus.HardFork.History as HardFork
+import qualified Ouroboros.Consensus.Ledger.SupportsMempool as Mempool
+import           Ouroboros.Consensus.Mempool (Mempool)
+import qualified Ouroboros.Consensus.Mempool as Mempool
+import qualified Ouroboros.Consensus.Mempool.Capacity as Mempool
+import           Ouroboros.Consensus.Util.IOLike (STM, atomically, retry)
+import           System.Random (randomIO)
+import           Test.Consensus.Mempool.Fairness.TestBlock
+import           Test.Tasty (TestTree, testGroup)
+import           Test.Tasty.ExpectedFailure (expectFail)
+import           Test.Tasty.HUnit (testCase, (@?), (@?=))
+import           Test.Util.TestBlock (testInitLedgerWithState)
+
+-- TODO: as boy-scout activity we might want to re-export symbols from
+-- Ouroboros.Consensus.Mempool.Capacity in Ouroboros.Consensus.Mempool
+
+tests :: TestTree
+tests = testGroup "Mempool fairness"
+                  [ expectFail $
+                    testCase "There is no substantial bias in added transaction sizes" $
+                              testTxSizeFairness TestParams { mempoolMaxCapacity =   100
+                                                            , smallTxSize        =     1
+                                                            , largeTxSize        =    10
+                                                            , nrOftxsToCollect   = 1_000
+                                                            , toleranceThreshold =     0.2 -- Somewhat arbitrarily chosen.
```

Comment:
Idea: QuickCheck's `checkCoverage` would provide some statistically sound tolerance values given a target distribution and generators. Perhaps something to keep in mind for later?